engine/schema/src/main/java/com/cloud/upgrade/dao/DatabaseAccessObject.java

```diff
@@ -26,6 +26,17 @@ public class DatabaseAccessObject {
 
     private static Logger s_logger = Logger.getLogger(DatabaseAccessObject.class);
 
+    public void addForeignKey(Connection conn, String tableName, String tableColumn, String foreignTableName, String foreignColumnName) {
+        String addForeignKeyStmt = String.format("CALL `cloud`.`IDEMPOTENT_ADD_FOREIGN_KEY`('%s', '%s', '%s', '%s')", tableName, tableColumn, foreignTableName, foreignColumnName);
```

Comment:
@harikrishna-patnala 
you can use `ALTER TABLE xxx ADD CONSTRAINT xxx` instead of cloud`.`IDEMPOTENT_ADD_FOREIGN_KEY`

Reply:
So you are saying we should not use the procedure in future also (I suppose you are saying to avoid errors with multiple mysql versions) ?